File: FJT.ReportViewer/MySqlDBModel/entity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace FJT.ReportViewer.MySqlDBModel
{
    public class entity
    {
        public entity()
        {
            reportmaster = new HashSet<reportmaster>();
        }

        [Key]
        public int entityID { get; set; }

        [StringLength(100)]
        public string entityName { get; set; }

        [StringLength(200)]
        public string remark { get; set; }

        public bool isActive { get; set; }

        public DateTime createdAt { get; set; }

        public DateTime? updatedAt { get; set; }

        public bool isDeleted { get; set; }

        public DateTime? deletedAt { get; set; }

        [StringLength(255)]
        public string createdBy { get; set; }

        [StringLength(255)]
        public string updatedBy { get; set; }

        [StringLength(255)]
        public string deletedBy { get; set; }

        public bool systemGenerated { get; set; }

        public int? columnView { get; set; }

        public int? entityStatus { get; set; }

        public int? createByRoleId { get; set; }

        public int? updateByRoleId { get; set; }

        public int? deleteByRoleId { get; set; }

        public virtual ICollection<reportmaster> reportmaster { get; set; }
    }
}

File: fjt.api/models/Entity.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const Entity = sequelize.define('Entity', {
        entityID: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        entityName: {
            type: Sequelize.STRING,
            allowNull: false,
            validate: {
                notEmpty: true,
                len: [1, 50]
            }
        },
        remark: {
            type: Sequelize.STRING,
            allowNull: true
        },
        isActive: {
            type: Sequelize.BOOLEAN,
            allowNull: true
            // defaultValue: true
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        systemGenerated: {
            type: Sequelize.BOOLEAN
        },
        columnView: {
            allowNull: true,
            type: Sequelize.INTEGER
        },
        entityStatus: {
            allowNull: true,
            type: Sequelize.INTEGER
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        displayName: {
            type: Sequelize.STRING,
            allowNull: true
        },
        jsonObjOfEnterprise: {
            type: Sequelize.JSON,
            allowNull: true
        },
        searchDisplayOrder: {
            type: Sequelize.DECIMAL(10, 5),
            allowNull: true
        },
        isDataEntity: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        }
    }, {
        paranoid: true,
        indexes: [{
            unique: true,
            fields: ['entityName']
        }],
        tableName: 'entity'
    });

    Entity.associate = (models) => {
        Entity.hasMany(models.DataElement, {
            as: 'dataElement',
            foreignKey: 'entityID'
        });
        Entity.hasMany(models.DynamicReportAccess, {
            foreignKey: 'refTransID',
            as: 'dynamicReportAccess'
        });
    };

    return Entity;
};
